test/Microsoft.TestPlatform.Common.UnitTests/Utilities/FakesUtilitiesTests.cs

```diff
@@ -39,5 +34,91 @@ public void FakesSettingsShouldBeNotGeneratedIfFakeConfiguratorAssemblyIsNotPres
+                Uri = new Uri(FakesUtilities.FakesMetadata.DataCollectorUriV2),
+                FriendlyName = FakesUtilities.FakesMetadata.FriendlyName,
+                IsEnabled = true,
+                Configuration = doc.FirstChild as XmlElement
+            };
+            FakesUtilities.InsertOrReplaceFakesDataCollectorNode(doc, dataCollectorNode2);
+
+            Assert.IsFalse(XmlRunSettingsUtilities.ContainsDataCollector(doc, FakesUtilities.FakesMetadata.DataCollectorUriV1));
+            Assert.IsTrue(XmlRunSettingsUtilities.ContainsDataCollector(doc, FakesUtilities.FakesMetadata.DataCollectorUriV2));
+        }
+
+        [TestMethod]
+        public void FakesDataCollectorSettingsShouldBeInserted()
+        {
+            string runSettingsXml = @"<RunSettings><RunConfiguration></RunConfiguration ></RunSettings>";
+            var doc = new XmlDocument();
+            using (var xmlReader = XmlReader.Create(
+                new StringReader(runSettingsXml),
+                new XmlReaderSettings() { CloseInput = true }))
+            {
+                doc.Load(xmlReader);
+            }
+
+            var dataCollectorNode2 = new DataCollectorSettings()
+            {
+                AssemblyQualifiedName = FakesUtilities.FakesMetadata.DataCollectorAssemblyQualifiedName,
+                Uri = new Uri(FakesUtilities.FakesMetadata.DataCollectorUriV2),
+                FriendlyName = FakesUtilities.FakesMetadata.FriendlyName,
+                IsEnabled = true,
+                Configuration = doc.FirstChild as XmlElement
+            };
+            FakesUtilities.InsertOrReplaceFakesDataCollectorNode(doc, dataCollectorNode2);
+            Assert.IsTrue(XmlRunSettingsUtilities.ContainsDataCollector(doc, FakesUtilities.FakesMetadata.DataCollectorUriV2));
+        }
+
+        [TestMethod]
+        public void OtherRunsettingsShouldNotBeChanged()
+        {
+            string runSettingsXml = @"<RunSettings><RunConfiguration><TargetFrameworkVersion>FrameworkCore10</TargetFrameworkVersion></RunConfiguration ></RunSettings>";
+            var doc = new XmlDocument();
+            using (var xmlReader = XmlReader.Create(
+                new StringReader(runSettingsXml),
+                new XmlReaderSettings() { CloseInput = true }))
+            {
+                doc.Load(xmlReader);
+            }
+
+            var dataCollectorNode2 = new DataCollectorSettings()
+            {
+                AssemblyQualifiedName = FakesUtilities.FakesMetadata.DataCollectorAssemblyQualifiedName,
+                Uri = new Uri(FakesUtilities.FakesMetadata.DataCollectorUriV2),
+                FriendlyName = FakesUtilities.FakesMetadata.FriendlyName,
+                IsEnabled = true,
+                Configuration = doc.CreateElement("Configuration")
+            };
+            FakesUtilities.InsertOrReplaceFakesDataCollectorNode(doc, dataCollectorNode2);
```

Comment:
>InsertOrReplaceFakesDataCollectorNode [](start = 27, length = 37)

Can an incorrect runsettings format be passed in to this API? Do we have checks for that?

Reply:
It is a possibility if called by an external user. Within testplatform, it isn't a possibility (other things would fail before this), and so the assumption has been that the runsettings passed in is sound.